test/e2e/metrics_test.go

```diff
@@ -287,3 +289,37 @@ func (tc *testContext) ensureMonitoringIsEnabled() error {
 
 	return nil
 }
+
+// testNodeResourceUsage ensures cpu/memory usage information is retrievable from Windows nodes
+func testNodeResourceUsage(t *testing.T) {
+	testCtx, err := NewTestContext()
+	require.NoError(t, err)
+
+	// Need at least one Windows node to run these tests, throwing error if this condition is not met
+	require.Greater(t, len(gc.nodes), 0, "test requires at least one Windows node to run")
+
+	for _, winNode := range gc.nodes {
+		t.Run(winNode.Name, func(t *testing.T) {
+			// check available resources
+			node, err := testCtx.client.K8s.CoreV1().Nodes().Get(context.TODO(), winNode.Name, metav1.GetOptions{})
+			require.NoErrorf(t, err, "could not retreive node object %s", winNode.Name)
+
+			cpuAvailable, isValidQuantity := node.Status.Allocatable.Cpu().AsInt64()
+			assert.Truef(t, isValidQuantity, "expected valid numeric quantity for node %s CPU usage", winNode.Name)
+			assert.NotZerof(t, cpuAvailable, "expected non-zero value for node %s CPU usage", winNode.Name)
+
+			memoryAvailable, isValidQuantity := node.Status.Allocatable.Memory().AsInt64()
+			assert.Truef(t, isValidQuantity, "expected valid numeric quantity for node %s memory usage", winNode.Name)
+			assert.NotZerof(t, memoryAvailable, "expected non-zero value for node %s memory usage", winNode.Name)
```

Comment:
Following  up to https://github.com/openshift/windows-machine-config-operator/pull/279#discussion_r636279251, I think these tests are already covered by `testWindowsPrometheusRules` which ensures the below metrics are non-zero.
 
```
node_cpu_utilisation 
node_cpu_info
node_cpu
memory_MemAvailable
memory_MemTotal
filesystem_size_bytes
node_filesystem_size_avail
node_network_receive_bytes_total
node_network_transmit_bytes_total
node_filesystem_free_bytes
```

Reply:
Just to document the reasoning behind keeping these tests (as no story associated with the work): 

> The CPU/memory stats tested here are served thru api-server by kubelet (which I assume collects via system calls as it is running on the hosts, same as windows_exporter). So if those tests fail then there are likely larger issues than just metrics errors - imo it would not hurt to add in the tests as it is testing a different path to the metrics (not queried through Prometheus like with windows_exporter)

Reply:
@saifshaikh48 for future reference, a good place for such documentation would be the commit message.